nxt/nxt_layer.py

```diff
@@ -474,6 +476,53 @@ def _world_descendants(self, return_type, include_implied=False):
             return node_table
         raise TypeError('Unsupported return type {}'.format(return_type))
 
+    def add_child_to_child_cache(self, parent_path, child_path, child_node):
+        if parent_path is nxt_path.WORLD:
+            return
+        children_nodes = []
+        children_paths = []
+        node_table = []
+        name_dict = {}
+        children_cache = self._cached_children.get(parent_path)
+        if children_cache is not None:
+            children_nodes = children_cache[LayerReturnTypes.Node][:]
+            children_paths = children_cache[LayerReturnTypes.Path][:]
+            node_table = children_cache[LayerReturnTypes.NodeTable][:]
+            name_dict = copy.copy(children_cache[LayerReturnTypes.NameDict])
+
+        children_nodes += [child_node]
+        children_paths += [child_path]
+        node_table += [[child_path, child_node]]
+        key = getattr(child_node, nxt_node.INTERNAL_ATTRS.NAME)
+        name_dict[key] = child_node
+
+        self._cached_children[parent_path] = {LayerReturnTypes.Node:
+                                                  children_nodes,
+                                              LayerReturnTypes.Path:
+                                                  children_paths,
+                                              LayerReturnTypes.NodeTable:
+                                                  node_table,
+                                              LayerReturnTypes.NameDict:
+                                                  name_dict}
+
+    def remove_child_to_child_cache(self, parent_path, child_path, child_node):
```

Comment:
seems like `remove_child_from_child_cache` is probably what this is meant to say.